util/src/main/scala/scala/scalanative/io/ByteBufferPool.scala

```diff
@@ -7,7 +7,7 @@ final class ByteBufferPool {
   private var buffers: List[ByteBuffer] = Nil
 
   private def alloc(): ByteBuffer = {
-    ByteBuffer.allocateDirect(64 * 1024 * 1024)
+    ByteBuffer.allocateDirect(8 * 1024 * 1024)
```

Comment:
There's also the codegen side, where we could investigate whether we can generate in one pass by reordering instructions to be aware of dependencies and whether we can render directly into a `CharBuffer` to not pay the `StringBuffer` => `byte[]` => `ByteBuffer` tax